src/language.h

```diff
@@ -155,30 +117,48 @@ struct StringParameters {
 
 	bool set_mode; ///< When not in set-mode, all parameters are cleared on first use of a Set function.
 	std::vector<StringParameterData> parms; ///< Parameters of the string.
+	int64 pluralize_count;                  ///< Value to use for selecting the plural form.
 };
 
-/**
- * Class for retrieving language strings.
- * @todo Implement me.
- */
-class Language {
+/** A string in one language with all its plural forms. */
+using PluralizedString = std::vector<const char*>;
+
+/** Contains all strings and the meta-information for one specific language. */
+class LanguageBundle {
+public:
+	LanguageBundle();
+	void Clear();
+	void InitMetaInfo(int index);
+
+	const char *GetSgText(StringID number) const;
+	const char *GetPlural(StringID number, int64 count) const;
+
+	std::vector<PluralizedString> values;  ///< Every known string with all its plural forms, indexed by #StringID.
+	                                       ///< Strings without a translation are \c nullptr. Memory is not owned.
+	const LanguageDefinition *metadata;    ///< This language's metadata.
+};
+
+/** Class for retrieving language strings. */
+class LanguageManager {
```

Comment:
`Manager` is a bit of a dangerous word, it's a breathes "everything fits in here" and they may end up as bloated classes that do everything.

Being aware of it is one defense against it. Another trick is to be a bit more precise in what it manages, thus reducing the scope of its name.

Reply:
Good point but I couldn't think of a more descriptive name, can you suggest one?